lib/miq_automation_engine/engine/miq_ae_engine.rb

```diff
@@ -329,6 +337,9 @@ def self.create_ae_attrs(attrs, name, vmdb_object, objects = [MiqServer.my_serve
       if !array_object.starts_with?("Array::")
         ae_attrs.delete(array_object)
       end
+      if !array_object.starts_with?("TagArray::") && array_attr_key.starts_with?("TagArray::")
+        ae_attrs.delete(array_object)
+      end
```

Comment:
I'm not sure what's being deleted here.  It looks like it's removing non-array objects.  If we need to do all of those conditions, you can combine them:

```ruby
if !array_object.starts_with?("Array::") || (!array_object.starts_with?("TagArray::") && array_attr_key.starts_with?("TagArray::"))
  ae_attrs.delete(array_object)
end
```